### ISSUE TITLE
Idea for enhancement: indicate subfolders shared with others

### ISSUE THREAD

**Issue by elpraga** [owncloud/android#1393]
Right not when someone shares a folder with me, the folder is marked as shared and so are any files and subfolders within.

On the other hand when I share a folder with someone, the folder is marked as shared, but the subfolders and files within are not.

It might be useful to be reminded that I am adding files to a folder that lies within a shared folder somewhere up the tree.

**Comment by tobiasKaminsky** [comment]
Do you have an idea how to show this indication?

**Comment by elpraga** [comment]
I will give it a thought. what would be the best way and I will post it here..

**Comment by tobiasKaminsky** [comment]
cc: @jancborchardt

**Comment by elpraga** [comment]
![folder_share_small](https://cloud.githubusercontent.com/assets/13814570/12176596/728a61c4-b568-11e5-8a98-6899446f6e4b.png)

The subfolders and files could be marked in this way. 
This way it would be clear that the folders and files lie within a folder we share, but it is not the folder that is being shared itself. It would also be distinguished from the folders and files others share with us. 

If it gets implemented, it would be important to show this distinction also in the folder chooser when files are being uploaded.

It might be interesting to implement this also in the web interface. What do Yo think?

**Comment by tobiasKaminsky** [comment]
cc @jancborchardt as he is the design guy ;)

I like the idea, but I am unsure if it 
- does not clutter the ui
- understandable by the users

But I do see the importance of this: Without this notifications one can put files into a shared folder and share them without knowing/wanting this.

**Comment by elpraga** [comment]
Yes, that was indeed the reason why I thought about this.

And as about cluttering, note that when opening subfolders shared with us all files are marked even more notably.

I tried to think about an option that would be informative, but as least invasive as possible (that is also why I choose white and not black for the share mark)

**Comment by elpraga** [comment]
@jancborchardt what do You think about this suggestion?

**Comment by jancborchardt** [comment]
Yeah, it should probably look the same whether you share something or the share comes from someone else. Otherwise it’s confusing. @elpraga do you want to take a stab at listing the cases and how it should look like?

Also taking into account how it currently is in the web interface and the mobile apps – it should be the same everywhere.

And a related issue is »Not possible to share current folder (need to display share icon in the breadcrumbs)« https://github.com/owncloud/core/issues/4333

**Comment by elpraga** [comment]
I will try to think about it, unfortunatelly, I am still without a computer.

**Comment by jesmrec** [comment]
How other clients behave:

- iOS: shows nothing
- Desktop: shows nothing
- Web (oC10): shows a small badge in the bottom right corner.

This is something to be escalated to product level. What's expected here?
